torch/_higher_order_ops/cond.py

```diff
@@ -156,7 +156,7 @@ def trace_cond(proxy_mode, func_overload, pred, true_fn, false_fn, operands):
 
     pre_dispatch = getattr(proxy_mode, "pre_dispatch", False)
 
-    with disable_proxy_modes_tracing():
+    with disable_proxy_modes_tracing(pre_dispatch=pre_dispatch):
```

Comment:
can we change this API to be something like proxy_mode.disable_proxy_modes_tracing(), and then delete `disable_proxy_modes_tracing`? this way it's more generic

Reply:
Will do as follow up! There are some use cases that use this API where there is no easy access to mode.